### PR TITLE
Bookmark List

### PR DESCRIPTION
**Pull request by garamond**
Provides a reading list / bookmarks / read later functionality

## To Do

- [x] Add action bar to profile
- [ ] Finalize names, routes
  - Leseliste?
  - Merkliste?
  - Lesezeichen?
- [x] Add to hamburger

#### Feed
image

#### Page
image

#### Profile
image

### Personal Feed «Lesezeichen» (currently at `/lesezeichen`)
image

### CODE REVIEW

File: components/ActionBar/Bookmark.js [hunk 1]
```diff
@@ -88,7 +88,13 @@ const addMutation = gql`
   ) {
     addDocumentToList(documentId: $documentId, listName: $listName) {
       id
-      name
+#      userListItems {
```
[the remark this answers is not in the document]
Comment (uxengine) [reply components/ActionBar/Bookmark.js]: It looks like `<Query query={query} partialRefetch={true}>` will fix the error (found here: https://github.com/apollographql/apollo-client/issues/2920#issuecomment-434719010).

However, it triggers a refetch and visible re-render when calling `addDocumentToList`. Anyway, I'll push a commit to move forward from there.

File: components/ActionBar/Bookmark.js [hunk 2]
```diff
@@ -90,7 +90,7 @@ const addMutation = gql`
       id
 #      userListItems {
```
Comment (uxengine) [inline components/ActionBar/Bookmark.js]: There's another issue maybe not related to, but at least triggered by `addDocumentToList`:

1. On localhost, navigate to the feed
2. Navigate to an article that is not bookmarked yet.
3. On the article's action bar, click on the bookmark icon to bookmark the article
4. Page return a 404 💥
5. Navigate back to the feed – the article was bookmarked successfully.

So the problem is the 404 which only occurs when bookmarking on the article (not when un-bookmarking).

File: components/ActionBar/Bookmark.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (tpreusse) [reply components/ActionBar/Bookmark.js]: fixed via https://github.com/orbiting/republik-frontend/pull/234/commits/1c6bfdf1993466d45b0c81b412870488a6560bba

File: components/ActionBar/Bookmark.js (hunk 2 above)
Comment (tpreusse) [reply components/ActionBar/Bookmark.js]: can repo, but maybe was related to https://github.com/orbiting/republik-frontend/pull/234/commits/e9721979d6bcac688faeb7fb22c7ef348fdbad61. Can you try to reproduce again

**Pull request merged**